crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -165,16 +209,23 @@ impl TransactionBlockEffects {
     // TODO: event_connection: EventConnection
 
     /// Base64 encoded bcs serialization of the on-chain transaction effects.
-    async fn bcs(&self) -> Result<Base64> {
+    async fn bcs(&self) -> Result<Option<Base64>> {
         let bytes = if let Some(stored) = self.tx_data.as_ref().left() {
-            stored.raw_effects.clone()
+            Some(stored.raw_effects.clone())
         } else {
-            bcs::to_bytes(&self.native)
-                .map_err(|e| Error::Internal(format!("Error serializing transaction effects: {e}")))
-                .extend()?
+            match self.effects {
+                Either::Left(ref native) => Some(
+                    bcs::to_bytes(native)
+                        .map_err(|e| {
+                            Error::Internal(format!("Error serializing transaction effects: {e}"))
+                        })
+                        .extend()?,
+                ),
+                Either::Right(_) => None,
```

Comment:
This is because `NativeTransactionEffects` is not easily fully derivable from `PreCommitTransactionBlockEffects`.